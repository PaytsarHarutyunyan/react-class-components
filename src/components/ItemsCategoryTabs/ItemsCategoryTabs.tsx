import React from 'react';
import classNames from 'classnames';
import Link from 'next/link';
import styles from '../../styles/ItemsCategoryTabs.module.css';
import { TABS } from '@/constants/commonConstants';
import { usePathname } from 'next/navigation';

const ItemsCategoryTabs: React.FC = () => {
    const pathname = usePathname();

    const drawButtons = () =>
        TABS.map((tab, index) => (
            <button
                className={classNames({
                    [styles.item]: true,
                    [styles.itemSelected]: pathname === tab.url,
                })}
                key={index}
            >
                <Link className={styles.itemLink} href={`${tab.url}?page=1`}>
                    {tab.name}
                </Link>
            </button>
        ));

    return <div className={styles.container}>{drawButtons()}</div>;
};

export default ItemsCategoryTabs;
