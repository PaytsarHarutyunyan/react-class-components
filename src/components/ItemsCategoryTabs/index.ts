import ItemsCategoryTabs from './ItemsCategoryTabs';
export default ItemsCategoryTabs;
