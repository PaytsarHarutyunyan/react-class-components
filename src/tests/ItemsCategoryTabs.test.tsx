import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ItemsCategoryTabs from '../components/ItemsCategoryTabs/ItemsCategoryTabs';

const mockPush = jest.fn();
jest.mock('next/navigation', () => ({
    useRouter: jest.fn(() => ({
        push: mockPush,
    })),
    redirect: jest.fn(),
    usePathname: jest.fn(() => '/people'),
    useSearchParams: jest.fn(() => ({ get: jest.fn() })),
}));

describe('ItemsCategoryTabs Component', () => {
    test('renders header buttons and handles click', () => {
        const { container } = render(<ItemsCategoryTabs />);

        const peopleButton = screen.getByText('People');
        const planetsButton = screen.getByText('Planets');

        expect(peopleButton).toBeInTheDocument();
        expect(planetsButton).toBeInTheDocument();

        fireEvent.click(planetsButton);

        expect(container).toMatchSnapshot();
    });
});
