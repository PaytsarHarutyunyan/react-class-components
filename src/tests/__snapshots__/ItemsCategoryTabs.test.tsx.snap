// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsCategoryTabs Component renders header buttons and handles click 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="item itemSelected"
    >
      <a
        class="itemLink"
        href="/people?page=1"
      >
        People
      </a>
    </button>
    <button
      class="item"
    >
      <a
        class="itemLink"
        href="/planets?page=1"
      >
        Planets
      </a>
    </button>
    <button
      class="item"
    >
      <a
        class="itemLink"
        href="/species?page=1"
      >
        Species
      </a>
    </button>
    <button
      class="item"
    >
      <a
        class="itemLink"
        href="/starships?page=1"
      >
        Starships
      </a>
    </button>
    <button
      class="item"
    >
      <a
        class="itemLink"
        href="/vehicles?page=1"
      >
        Vehicles
      </a>
    </button>
  </div>
</div>
`;
